homeassistant/components/onewire/binary_sensor.py

```diff
@@ -0,0 +1,132 @@
+"""Support for 1-Wire binary sensors."""
+import os
+
+from homeassistant.components.binary_sensor import BinarySensorEntity
+from homeassistant.const import CONF_TYPE
+
+from .const import CONF_TYPE_OWSERVER, DOMAIN, SENSOR_TYPE_SENSED
+from .onewire_entities import OneWireProxy
+from .onewirehub import OneWireHub
+
+DEVICE_BINARY_SENSORS = {
+    # Family : { path, sensor_type }
+    "12": [
+        {
+            "path": "sensed.A",
+            "name": "Sensed A",
+            "type": SENSOR_TYPE_SENSED,
+            "default_disabled": True,
+        },
+        {
+            "path": "sensed.B",
+            "name": "Sensed B",
+            "type": SENSOR_TYPE_SENSED,
+            "default_disabled": True,
+        },
+    ],
+    "29": [
+        {
+            "path": "sensed.0",
+            "name": "Sensed 0",
+            "type": SENSOR_TYPE_SENSED,
+            "default_disabled": True,
+        },
+        {
+            "path": "sensed.1",
+            "name": "Sensed 1",
+            "type": SENSOR_TYPE_SENSED,
+            "default_disabled": True,
+        },
+        {
+            "path": "sensed.2",
+            "name": "Sensed 2",
+            "type": SENSOR_TYPE_SENSED,
+            "default_disabled": True,
+        },
+        {
+            "path": "sensed.3",
+            "name": "Sensed 3",
+            "type": SENSOR_TYPE_SENSED,
+            "default_disabled": True,
+        },
+        {
+            "path": "sensed.4",
+            "name": "Sensed 4",
+            "type": SENSOR_TYPE_SENSED,
+            "default_disabled": True,
+        },
+        {
+            "path": "sensed.5",
+            "name": "Sensed 5",
+            "type": SENSOR_TYPE_SENSED,
+            "default_disabled": True,
+        },
+        {
+            "path": "sensed.6",
+            "name": "Sensed 6",
+            "type": SENSOR_TYPE_SENSED,
+            "default_disabled": True,
+        },
+        {
+            "path": "sensed.7",
+            "name": "Sensed 7",
+            "type": SENSOR_TYPE_SENSED,
+            "default_disabled": True,
+        },
+    ],
+}
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up 1-Wire platform."""
+    # Only OWServer implementation works with binary sensors
+    if config_entry.data[CONF_TYPE] == CONF_TYPE_OWSERVER:
+        onewirehub = hass.data[DOMAIN][config_entry.unique_id]
```

Comment:
Please separate words with underscore in variable and module names. Change to `onewire_hub`.